### PR TITLE
Create Signup Screen UI

### PR DESCRIPTION
**Pull request by cemdemir-dev**
This PR adds Signup screen and Signup validations.

### CODE REVIEW

**Review by safaorhan — CHANGES_REQUESTED** [review]
Congratz for the first ever PR! 🥳 

If you can address my comments we can take a second look together!

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <application
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
-        android:label="@string/app_name"
+        android:label="Sign up"
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/AndroidManifest.xml:8]: Done!

File: app/src/main/java/com/example/cactusnotes/SignupActivity.kt [hunk 2]
```diff
@@ -2,10 +2,86 @@ package com.example.cactusnotes
...
 import androidx.appcompat.app.AppCompatActivity
+import com.example.cactusnotes.databinding.ActivitySignupBinding
 
 class SignupActivity : AppCompatActivity() {
+    lateinit var binding: ActivitySignupBinding
+
+    fun String.containsLowerCase() = any { it.isLowerCase() }
+    fun String.containsUpperCase() = any { it.isUpperCase() }
+    fun String.containsDigit() = any { it.isDigit() }
+    fun Char.isSpecial() = (!isDigit() && !isLetter())
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/java/com/example/cactusnotes/SignupActivity.kt:13]: Done!

File: app/src/main/java/com/example/cactusnotes/SignupActivity.kt [hunk 3]
```diff
@@ -2,10 +2,86 @@ package com.example.cactusnotes
...
+    fun String.containsSpecChar() = any { it.isSpecial() }
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/java/com/example/cactusnotes/SignupActivity.kt:14]: Done!

File: app/src/main/java/com/example/cactusnotes/SignupActivity.kt [hunk 4]
```diff
@@ -2,10 +2,86 @@ package com.example.cactusnotes
...
+        binding.signUpButton.setOnClickListener {
+            val username = binding.usernameInputLayout.editText!!.text.toString()
+            val password = binding.passwordInputLayout.editText!!.text.toString()
+            val email = binding.emailInputLayout.editText!!.text.toString()
+
+            // Password checking
+            when {
+                !password.containsLowerCase() || !password.containsDigit() ||
+                        !password.containsUpperCase() || !password.containsSpecChar() -> {
+                    binding.passwordInputLayout.error = getString(R.string.password_special_char)
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/java/com/example/cactusnotes/SignupActivity.kt:30]: Done!

File: app/src/main/java/com/example/cactusnotes/SignupActivity.kt [hunk 5]
```diff
@@ -2,10 +2,86 @@ package com.example.cactusnotes
...
+                    binding.passwordInputLayout.error = null
+                    binding.passwordInputLayout.isErrorEnabled = false
+                }
+            }
+            // Email checking
+            when {
+                email == "" -> {
+                    binding.emailInputLayout.error = getString(R.string.e_mail_required)
+                }
+                !email.contains("@") && (!email.contains(".")) -> {
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/java/com/example/cactusnotes/SignupActivity.kt:48]: Done!

File: app/src/main/java/com/example/cactusnotes/SignupActivity.kt [hunk 6]
```diff
@@ -2,10 +2,86 @@ package com.example.cactusnotes
...
+                username == "" -> {
+                    binding.usernameInputLayout.error = getString(R.string.username_is_required)
+                }
+                username.length < 3 -> {
+                    binding.usernameInputLayout.error = getString(R.string.username_too_short)
+                }
+                username.length > 19 -> {
+                    binding.usernameInputLayout.error = getString(R.string.username_too_long)
+                }
+                !username.all { it.isLowerCase() || it.isDigit() || username.contains("_") } -> {
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/java/com/example/cactusnotes/SignupActivity.kt:76]: Done!

File: app/src/main/res/layout/activity_signup.xml [hunk 7]
```diff
@@ -1,18 +1,77 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:padding="18dp"
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/res/layout/activity_signup.xml:6]: Done!

File: app/src/main/res/layout/activity_signup.xml [hunk 8]
```diff
@@ -1,18 +1,77 @@
...
+        tools:ignore="InvalidId">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/username" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/res/layout/activity_signup.xml:39]: Done!

File: app/src/main/java/com/example/cactusnotes/SignupActivity.kt [hunk 9]
```diff
@@ -2,10 +2,86 @@ package com.example.cactusnotes
...
+        binding = ActivitySignupBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.signUpButton.setOnClickListener {
+            val username = binding.usernameInputLayout.editText!!.text.toString()
+            val password = binding.passwordInputLayout.editText!!.text.toString()
+            val email = binding.emailInputLayout.editText!!.text.toString()
+
+            // Password checking
+            when {
```
[the remark this answers is not in the document]
Comment (cemdemir-dev) [reply app/src/main/java/com/example/cactusnotes/SignupActivity.kt:27]: Done!

**Pull request merged**